src/main/java/picard/analysis/CollectQualityYieldMetricsSNVQ.java

```diff
@@ -0,0 +1,548 @@
+            "<a href='https://www.broadinstitute.org/gatk/guide/article?id=6329'>GATK Dictionary entry</a>. " +
+            "<br />" +
+
+            "<h4>Usage Example:</h4>" +
+            "<pre>" +
+            "java -jar picard.jar CollectSNVQualityYieldMetrics \\<br /> " +
+            "      I=input.bam \\<br /> " +
+            "      O=quality_yield_metrics.txt \\<br />" +
+            "</pre>" +
+            "Please see " +
+            "<a href='https://broadinstitute.github.io/picard/picard-metric-definitions.html#CollectSNVQualityYieldMetrics.QualityYieldMetrics'>" +
+            "the QualityYieldMetrics documentation</a> for details and explanations of the output metrics." +
+            "<hr />";
+
+    @Argument(shortName = StandardOptionDefinitions.USE_ACTUAL_BASE_QUALITIES_SHORT_NAME,
+            doc = "Use the actual base quality (QUAL) scores " +
+                    "as inputs instead of the quality scores in the BQ field.")
+    public boolean USE_ACTUAL_ORIGINAL_QUALITIES = false;
+
+    @Argument(doc = "If true, include bases from secondary alignments in metrics. Setting to true may cause double-counting " +
+            "of bases if there are secondary alignments in the input file.")
+    public boolean INCLUDE_SECONDARY_ALIGNMENTS = false;
+
+    @Argument(doc = "If true, include bases from supplemental alignments in metrics. Setting to true may cause double-counting " +
+            "of bases if there are supplemental alignments in the input file.")
+    public boolean INCLUDE_SUPPLEMENTAL_ALIGNMENTS = false;
+
+    @Argument(doc = "Determines whether to include the base quality histogram in the metrics file.")
+    public boolean INCLUDE_BQ_HISTOGRAM = false;
+
+    /**
+     * Ensure that we get all reads regardless of alignment status.
+     */
+    @Override
+    protected boolean usesNoRefReads() {
+        return true;
+    }
+
+    @Override
+    protected void setup(final SAMFileHeader header, final File samFile) {
+        IOUtil.assertFileIsWritable(OUTPUT);
+        this.collector = new QualityYieldMetricsCollector(USE_ACTUAL_ORIGINAL_QUALITIES, INCLUDE_SECONDARY_ALIGNMENTS, INCLUDE_SUPPLEMENTAL_ALIGNMENTS);
+    }
+
+    @Override
+    protected void acceptRead(final SAMRecord rec, final ReferenceSequence ref) {
+        this.collector.acceptRecord(rec, ref);
+    }
+
+    @Override
+    protected void finish() {
+        final MetricsFile<QualityYieldMetrics, Integer> metricsFile = getMetricsFile();
+        this.collector.finish();
+        this.collector.addMetricsToFile(metricsFile);
+        if ( INCLUDE_BQ_HISTOGRAM ) {
+            metricsFile.addHistogram(qualityHistogram);
+            metricsFile.addHistogram(snvqHistogram);
+            this.collector.addHistograms(metricsFile);
+        }
+        metricsFile.write(OUTPUT);
+    }
+
+    public class QualityYieldMetricsCollector {
+        // If true, include bases from secondary alignments in metrics. Setting to true may cause double-counting
```

Comment:
The docstring does not match the parameter I think

Reply:
Now aligned